### PR TITLE
Fixing error for OOP isUndefined.

### LINKED ISSUE #208: Underscore.js OOP style doesn't report as negative
The rules only report negatives when using Undercore's functional style. 
When using [OOP Style](https://underscorejs.org/#oop) it doesn't work:

image

### PR DESCRIPTION
**Pull request by cjtaylor1990**
I believed I was able to fix it. While I was doing console.log's I realized that the callee objects for the two different lines (_(myVar).isUndefined() v.s. _.isUndefined(myVar)) didn't match. The former had a structure such that:

callee.object.callee.name = '_'

rather than,
callee.object.name = '_'

This means that in certain odd cases, you'll have to go one level deeper into the callee object to find the name you are looking for. I've changed the assigning of objectName accordingly.

Fixed #208

### CODE REVIEW

**Comment by stevemao** [comment]
Thanks @cjtaylor1990 
Could you please add a unit test for this?

**Comment by cjtaylor1990** [comment]
@stevemao Absolutely. I presume the relevant location would be test/unit/all.js ?

**Comment by cjtaylor1990** [comment]
@cylim Just added it.

**Comment by stevemao** [comment]
> I presume the relevant location would be test/unit/all.js ?

yes

> Can you add a fixed #<issues number>

it needs to be in the commit message instead of the description f this PR

**Comment by cjtaylor1990** [comment]
@stevemao Okay, I'll try to work on that a bit then. I've never written unit tests in JS before, but there's plenty of documentation about and I'm sure it won't be too difficult. I'll also put the issue # in the commit with the unit test.

**Comment by stevemao** [comment]
Thanks buddy!

**Comment by cjtaylor1990** [comment]
@stevemao. I added in a unit test into test/unit/all.js for isUndefined. Should I also include a test in tests/lib/rules/app.js? The latter seems to be testing the eslinter rules while the former seems to be testing the lodash methods against the native methods.

**Comment by stevenvachon** [comment]
@cjtaylor1990 this is causing:
```
TypeError: Cannot read property 'callee' of undefined
```

**Comment by cjtaylor1990** [comment]
@stevenvachon Okay, I'll fix that.

**Comment by stevenvachon** [comment]
* My [eslint config](https://github.com/stevenvachon/broken-link-checker/blob/master/scripts/eslintrc.cjs.js)
* The [file](https://github.com/stevenvachon/broken-link-checker/blob/master/lib/cli.js) causing the errror

**Comment by cjtaylor1990** [comment]
@stevenvachon Could you do me a favor? I cannot quite reproduce what you're experiencing, but I think I know how to fix your issue. Essentially, there was a hole in the logic of how I wrote things and it can fail in very specific situations.

In ./node_modules/eslint-plugin-you-dont-need-lodash-underscore/lib/rules/all.js, on line 28, you'll see:

`const objectName = callee.name || (callee.object && callee.object.name) || (callee.object.callee && callee.object.callee.name);`

Try changing it to:
`const objectName = callee.name || (callee.object && callee.object.name) || (callee.object.callee && callee.object.callee.name);`

That might do it. Tell me how it goes.

**Comment by cjtaylor1990** [comment]
@stevenvachon It might also be helpful to open up a new Issue and reference this PR (#219). I think the maintainers would prefer that just to keep everything organized. Sorry that I didn't point you in that direction sooner.

**Comment by stevenvachon** [comment]
@cjtaylor1990 Yes, that fixes it, but you'll need a test to prevent this from happening in the future.

**Comment by cjtaylor1990** [comment]
@stevenvachon Of course. Thank you for pointing out the issue.

**Comment by cjtaylor1990** [comment]
@stevemao I've done the fix for the issue mentioned by @stevenvachon and committed it to my branch on my clone. Where would you suggest putting the test since the it is a test of lib/rules/all.js rather than comparing native methods to that of _/Lodash methods?

Sorry, as I've said before, rather new to writing JS tests and I don't want to mess up the structure of the project.

**Comment by stevemao** [comment]
What about https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore/blob/master/tests/lib/rules/all.js ? Feel free to make the structure better.

**Comment by cjtaylor1990** [comment]
@stevenvachon  I actually couldn't ever reproduce the TypeError you mentioned, but I did find another TypeError when running eslint with the old script (the un-fixed one) on ./lib/public/HtmlChecker.js. The issue was in line 30 of HtmlChecker where you're using a using a function to define another function, which you then return this new function, evaluated with the result argument:

`link: function(result)
		{
			maybeCallback(thisObj.handlers.link)(result);
		},
end: function()`

So, to put it simply: it was having problems with functions that return functions. I'll make sure that this is scenario is put inside of the test suite.
